### PR TITLE
docs: introduce adrs

### PR DESCRIPTION
**Pull request by richardcase**
**What this PR does / why we need it**:

As a lightweight way to document decisions, we will use ADRs that will be stored in the repo.

This PR includes a template and an ADR to propose that we use ADRs.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits into logical changes
- [x] includes documentation

### CODE REVIEW

File: docs/adr/0000-template.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# 0. Title
```
Comment (furkatgofurov7) [inline docs/adr/0000-template.md:1]:
```suggestion
# Title
```

File: docs/adr/0000-template.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+<!-- A short and clear title which is prefixed with the ADR number -->
+
+* Status: [proposed | rejected | accepted | deprecated | … | superseded by [ADR-0005](0005-example.md)] <!-- mandatory -->
+* Date: 2020-10-29 [YYY-MM-DD - date of the decision] <!-- mandatory -->
+* Authors: [list of GitHub handles for the authors]
+* Deciders: [list of GitHub handles for those that made the decision]  <!-- mandatory -->
+
+## Context
+<!-- What is the context of the decision and whats the motivation -->
```
Comment (furkatgofurov7) [inline docs/adr/0000-template.md:10]:
```suggestion
<!-- What is the context of the decision and what's the motivation -->
```

**Review by furkatgofurov7 — APPROVED** [review]

**Review by Danil-Grigorev — APPROVED** [review]

**Review by alexander-demicev — DISMISSED** [review]

**Comment by richardcase** [comment]
Committed the changes suggested by @furkatgofurov7

**Comment by richardcase** [comment]
Merging based on the 4 approvals

**Pull request merged**